src/main/java/frc/robot/subsystems/climberSubsystem/ClimberSubsystem.java

```diff
@@ -0,0 +1,176 @@
+  }
+
+  /**
+   * Apply a currentlimit
+   *
+   * @param currentLimit the limit in amps to set
+   */
+  public void setCurrentLimit(int currentLimit) {
+    setPIDState(PIDState.CURRENT);
+    leftArmMotor.setSmartCurrentLimit(currentLimit);
+    rightArmMotor.setSmartCurrentLimit(currentLimit);
+  }
+
+  public void zeroAtCurrentPosition() {
+    leftArmMotor.getEncoder().setPosition(0);
+    leftArmMotor.getEncoder().setPosition(0);
+  }
+
+  /**
+   * Any command that uses this needs to clean up after self.
+   * As in reset min and max to -1 and  1
+   * @param max the maximum output (Physically 1)
+   * @param min the minimum output (Physically -1)
+   */
+  public void setOutputLimits(double max, double min) {
+    leftArmMotor.getPIDController().setOutputRange(min,max);
+    rightArmMotor.getPIDController().setOutputRange(min,max);
+  }
+
+  public Position getPosition() {
+    return new Position(
+        leftArmMotor.getEncoder().getPosition(), rightArmMotor.getEncoder().getPosition());
+  }
+
+  public CurrentVelocity getVelocity() {
+    return new CurrentVelocity(
+        leftArmMotor.getEncoder().getVelocity(), rightArmMotor.getEncoder().getVelocity());
+  }
+
+  public AppliedCurrent getCurrent() {
+    return new AppliedCurrent(leftArmMotor.getOutputCurrent(), rightArmMotor.getOutputCurrent());
+  }
+
+  public void requestPosition(double climbPosition) {
+    setPIDState(PIDState.POSITION);
+    leftArmMotor.getPIDController().setReference(climbPosition, ControlType.kPosition);
+    rightArmMotor.getPIDController().setReference(climbPosition, ControlType.kPosition);
+  }
+
+  private void setPIDState(PIDState state){
+    if(state == pidState) return;
+    configPID(rightArmMotor.getPIDController(), state.getPID());
+    configPID(leftArmMotor.getPIDController(), state.getPID());
+  }
+
+  private static void configPID(SparkPIDController pidController, double[] pid){
+    if(pid.length != 3) throw new RuntimeException("Incorrect number of PID numbers");
+    pidController.setP(pid[0]);
+    pidController.setI(pid[1]);
+    pidController.setD(pid[2]);
+  }
+
+  @Override
+  public void periodic() {}
+
+  public record DifferentialDemand(double left, double right) {}
+
+  public record Position(double left, double right) {
+
+    public boolean isLessThen(double arg) {
+      return left < arg && right < arg;
+    }
+  }
+
+  public record AppliedCurrent(double left, double right) {
+    public boolean greater(double arg) {
+
+      return left > arg && right > arg;
+    }
+  }
+
+  public record CurrentVelocity(double left, double right) {
+    public boolean isZero(double epsilion) {
+      return Math.abs(left) < epsilion && Math.abs(right) < epsilion;
+    }
+  }
+
+  private enum PIDState {
+    CURRENT(0),
+    POSITION(1);
+    private int id;
+    private PIDState(int id){
+      this.id = id;
+    }
+
+    public double[] getPID(){
+      switch(this.id){
+        case 0:
```

Comment:
Should we be using the enum here?

Reply:
can't. This enum itself represents what is being switched on here.